nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/test/java/org/apache/nifi/processors/azure/storage/ITPutAzureDataLakeStorage.java

```diff
@@ -32,26 +49,223 @@
 
     @Before
     public void setUp() {
-        runner.setProperty(PutAzureDataLakeStorage.FILE, TEST_FILE_NAME);
+        runner.setProperty(PutAzureDataLakeStorage.DIRECTORY, DIRECTORY);
+        runner.setProperty(PutAzureDataLakeStorage.FILE, FILE_NAME);
+    }
+
+    @Test
+    public void testPutFileToExistingDirectory() throws Exception {
+        fileSystemClient.createDirectory(DIRECTORY);
+
+        runProcessor(FILE_DATA);
+
+        assertSuccess(DIRECTORY, FILE_NAME, FILE_DATA);
+    }
+
+    @Test
+    public void testPutFileToNonExistingDirectory() throws Exception {
+        runProcessor(FILE_DATA);
+
+        assertSuccess(DIRECTORY, FILE_NAME, FILE_DATA);
+    }
+
+    @Ignore
+    // DataLakeFileClient.getFileUrl() returns "dir1%2Fdir2%2Fdir3%2Fdir4" for "dir1/dir2/dir3/dir4"
+    // seems to be a bug in the Azure lib
```

Comment:
@MuazmaZ Do you think it is a bug or works as expected?
I would expect
`https://mystorageaccount.dfs.core.windows.net/myfilesystem/dir1/dir2/dir3/dir4/file1`
(as displayed on the Azure Portal)
instead of
`https://mystorageaccount.dfs.core.windows.net/myfilesystem/dir1%2Fdir2%2Fdir3%2Fdir4/file1`

Reply:
Hi @turcsanyip Yes, this is expected behavior. The builder/client internally url encodes the URL provided. If you need access to the non-url encoded url, you can use the Utility.urlDecode function in the common library.

Reply:
Hi @turcsanyip I usually run integration tests with 'mvn verify -Pintegration-tests' command at the root of nifi-azure-bundle after configuring test values in $HOME/azure-credentials.PROPERTIES.

Can you tell me what else I need to do in order to run your integration tests?

Reply:
@MuazmaZ Thanks for the clarification. I modified the test and removed the `@Ignore`.

@sjyang18 Do you have some test failures or it does not work at all?
`mvn verify -Pintegration-tests` should work with `$HOME/azure-credentials.PROPERTIES`.
There were some failing tests in `ITFetchAzureDataLakeStorage` and `ITDeleteAzureDataLakeStorage`. I `@Ignore`-d them for now, they will be fixed soon in other jira tickets.
Please send the error logs if it does not work for you now.

Reply:
+1 vote.  I saw the exception in the output, and it turns out that your test case expects failures (assertFailure).